docs/charmil_commands.md

```diff
@@ -0,0 +1,26 @@
+## Charmil Commands
+Charmil provides prebuilt commands for developers to use when testing their application. These commands are opt-in and are only added when the application is built with a specific tag.
+
+To add charmil commands to your root command you can call the `CharmilCommands()` function. If calling this function does not populate `err` then add the commands to your root command.  
```

Comment:
I love this docs however we do not have clear example for the commands like this right now or CharmilCommands() in core. Thinking how to keep this work so we can actually implement things this way.

One of the ideas would be to push this docs to master in separate PR and create placeholder for charmil commands. 
Then rest of the team will simply reuse that placeholder to inject commands as they build features. 
This will be amazing for us to have docs and end user API first and will solve challenges on "HOW TO USE THIS DAM THING :) "